core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,203 +16,241 @@
-        if (successCallback) {
-          successCallback();
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _changeStoryPublicationStatus(
+      storyId: string,
+      newStoryStatusIsPublic: boolean,
+      successCallback: (value: number) =>void,
+      errorCallback: (reason: string) => void): void {
+    const storyPublishUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.STORY_PUBLISH_URL_TEMPLATE, {
+        story_id: storyId
       });
+    var putData = {
+      new_story_status_is_public: newStoryStatusIsPublic
     };
+    // eslint-disable-next-line max-len
+    this.http.put(storyPublishUrl, putData, { observe: 'response'}).toPromise().then(
+      response => successCallback(response.status),
+      errorResponse => errorCallback(errorResponse.error.error));
+  }
 
-    var _validateExplorations = function(
-        storyId, expIds, successCallback, errorCallback) {
-      var validateExplorationsUrl = UrlInterpolationService.interpolateUrl(
-        VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
+  private _validateExplorations = function(
+      storyId: string,
+      expIds: string[],
+      successCallback: (value?: string | PromiseLike<Object>) =>void,
+      errorCallback: (reason?: string | PromiseLike<Object>) => void): void {
+    const validateExplorationsUrl =
+      this.urlInterpolationService.interpolateUrl(
+        StoryDomainConstants.VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
           story_id: storyId
         });
 
-      $http.get(validateExplorationsUrl, {
+    this.http.get(
+      validateExplorationsUrl, {
         params: {
           comma_separated_exp_ids: expIds.join(',')
         }
-      }).then(function(response) {
-        if (successCallback) {
-          successCallback(response.data.validation_error_messages);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+      }
+    ).toPromise().then(
+      response => successCallback(response.status),
+      errorResponse => errorCallback(errorResponse.error.error));
+  };
+
+  private _deleteStory = function(
+      storyId: string,
+      successCallback: (value?: string | PromiseLike<Object>) =>void,
+      errorCallback: (reason?: string | PromiseLike<Object>) => void): void {
+    const storyDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
-    };
 
-    var _deleteStory = function(
-        storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
-      $http['delete'](storyDataUrl).then(function(response) {
+    this.http.request(
+      'delete',
```

Comment:
why not this.http.delete ?

Reply:
It gives lint error with this.http.delete

Reply:
Use eslint disable statement to suppress that error

Reply:
done